File: star-syncplayer-client/src/main/java/top/tonydon/syncplayer/util/AlertUtils.java
package top.tonydon.syncplayer.util;

import javafx.application.Platform;
import javafx.scene.control.Alert;
import javafx.stage.Modality;
import javafx.stage.Window;

/**
 * 提示框工具类
 */
public class AlertUtils {

    public static void confirmation(String head, String context, Window window) {
        alter(head, context, window, Alert.AlertType.CONFIRMATION);
    }

    public static void confirmation(String head, Window window) {
        confirmation(head, null, window);
    }

    public static void none(String head, String context, Window window) {
        alter(head, context, window, Alert.AlertType.NONE);
    }

    public static void none(String head, Window window) {
        none(head, null, window);
    }

    public static void error(String head, String context, Window window) {
        alter(head, context, window, Alert.AlertType.ERROR);
    }

    public static void error(String head, Window window) {
        error(head, null, window);
    }

    public static void information(String head, String context, Window window) {
        alter(head, context, window, Alert.AlertType.INFORMATION);
    }

    public static void information(String head, Window window) {
        information(head, null, window);
    }

    public static void warning(String head, String context, Window window) {
        alter(head, context, window, Alert.AlertType.WARNING);
    }

    public static void warning(String head, Window window) {
        warning(head, null, window);
    }

    private static void alter(String head, String context, Window window, Alert.AlertType type) {
        Platform.runLater(() -> {
            Alert alert = new Alert(type);
            alert.setHeaderText(head);
            alert.setContentText(context);
            alert.initModality(Modality.WINDOW_MODAL);
            alert.initOwner(window);
            alert.show();
        });
    }
}

File: star-syncplayer-client/src/main/java/top/tonydon/syncplayer/util/AlertUtils.kt
package top.tonydon.syncplayer.util

import javafx.application.Platform
import javafx.scene.control.Alert
import javafx.scene.control.Alert.AlertType
import javafx.stage.Modality
import javafx.stage.Window

/**
 * 提示框工具类
 */
object AlertUtils {
    fun confirmation(head: String, context: String?, window: Window) {
        alter(head, context, window, AlertType.CONFIRMATION)
    }

    fun confirmation(head: String, window: Window) {
        confirmation(head, null, window)
    }

    fun none(head: String, context: String?, window: Window) {
        alter(head, context, window, AlertType.NONE)
    }

    fun none(head: String, window: Window) {
        none(head, null, window)
    }

    fun error(head: String, context: String?, window: Window) {
        alter(head, context, window, AlertType.ERROR)
    }

    fun error(head: String, window: Window) {
        error(head, null, window)
    }

    fun information(head: String, context: String?, window: Window) {
        alter(head, context, window, AlertType.INFORMATION)
    }

    fun information(head: String, window: Window) {
        information(head, null, window)
    }

    fun warning(head: String, context: String?, window: Window) {
        alter(head, context, window, AlertType.WARNING)
    }

    fun warning(head: String, window: Window) {
        warning(head, null, window)
    }

    private fun alter(head: String, context: String?, window: Window, type: AlertType) {
        Platform.runLater {
            val alert = Alert(type)
            alert.headerText = head
            alert.contentText = context
            alert.initModality(Modality.WINDOW_MODAL)
            alert.initOwner(window)
            alert.show()
        }
    }
}
